src/mpid/ch4/include/mpidpost.h

```diff
@@ -28,9 +28,9 @@ MPL_STATIC_INLINE_PREFIX void MPID_Request_free_hook(MPIR_Request * req)
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPID_REQUEST_FREE_HOOK);
 
     int vci = MPIDI_Request_get_vci(req);
-    /* Increment MPIDI_global.progress_counts[vci]. */
-    int count = MPL_atomic_relaxed_load_int(&MPIDI_global.progress_counts[vci]);
-    MPL_atomic_relaxed_store_int(&MPIDI_global.progress_counts[vci], count + 1);
+    /* Increment MPIDI_global.vci[vci].vci.progress_count. */
+    int count = MPL_atomic_relaxed_load_int(&MPIDI_global.vci[vci].vci.progress_count);
+    MPL_atomic_relaxed_store_int(&MPIDI_global.vci[vci].vci.progress_count, count + 1);
```

Comment:
Instead of have 3 consecutive `vci`, you can use the macro: `MPIDI_VCI(vci).progress_count`.

Reply:
Thanks for reviewing this PR, @hzhou! This macro makes lines much shorter!
I replaced all `MPIDI_global.vci[XXX].vci.progress_count` with `MPIDI_VCI(XXX).progress_count`.